dask/dataframe/io/io.py

```diff
@@ -244,61 +244,75 @@ def from_pandas(
     Raises
     ------
     TypeError
-        If something other than a ``pandas.DataFrame`` or ``pandas.Series`` is
-        passed in.
+        If something other than a ``pandas.DataFrame`` or ``pandas.Series`` is passed in.
+    ValueError
+        If passed a DataFrame with duplicate column names.
+    NotImplementedError
+        If passed either multi-indexed or null-indexed numeric data.
 
     See Also
     --------
     from_array : Construct a dask.DataFrame from an array that has record dtype
     read_csv : Construct a dask.DataFrame from a CSV file
     """
+    # Check data
+    if not has_parallel_type(data):
+        raise TypeError("Input must be a pandas DataFrame or Series.")
+
+    # Check index
     if isinstance(getattr(data, "index", None), pd.MultiIndex):
         raise NotImplementedError("Dask does not support MultiIndex Dataframes.")
 
-    if not has_parallel_type(data):
-        raise TypeError("Input must be a pandas DataFrame or Series.")
+    # Check data has no duplicate column names
+    # Even if this is eventually checked during construction, it's faster to check it at this point
+    if is_dataframe_like(data):
+        # Here it's assumed that a named index was originally a column, so we prevent a name clash
+        DataFrame._prevent_dupl_col_names(data, allow_index_name_in_cols=False)
 
+    # Check we have either chunksize or npartitions to know how to partition
     if (npartitions is None) == (chunksize is None):
         raise ValueError("Exactly one of npartitions and chunksize must be specified.")
-
-    nrows = len(data)
-
-    if chunksize is None:
-        if not isinstance(npartitions, int):
-            raise TypeError(
-                "Please provide npartitions as an int, or possibly as None if you specify chunksize."
-            )
-    elif not isinstance(chunksize, int):
+    if chunksize is None and not isinstance(npartitions, int):
         raise TypeError(
-            "Please provide chunksize as an int, or possibly as None if you specify npartitions."
+            "Please provide npartitions as an int, or set it to None if you specify chunksize."
```

Comment:
Did this actually change, or was this message wrong?

Reply:
I changed it to make more clear to the user that only 1 of `npartitions` and `chunksize` should be provided (in a XOR-like way).